src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts

```diff
@@ -80,33 +94,58 @@ export class RelationshipEffects {
    */
   @Effect({ dispatch: false }) updateNameVariantsActions$ = this.actions$
     .pipe(
-      ofType(RelationshipActionTypes.UPDATE_RELATIONSHIP),
-      map((action: UpdateRelationshipAction) => {
-          const { item1, item2, relationshipType, nameVariant } = action.payload;
+      ofType(RelationshipActionTypes.UPDATE_NAME_VARIANT),
+      map((action: UpdateRelationshipNameVariantAction) => {
+          const { item1, item2, relationshipType, submissionId, nameVariant } = action.payload;
           const identifier: string = this.createIdentifier(item1, item2, relationshipType);
           const inProgress = hasValue(this.debounceMap[identifier]);
           if (inProgress) {
             this.nameVariantUpdates[identifier] = nameVariant;
           } else {
-            this.relationshipService.updateNameVariant(item1, item2, relationshipType, nameVariant)
-              .pipe(getSucceededRemoteData())
-              .subscribe();
+            this.relationshipService.updateNameVariant(item1, item2, relationshipType, nameVariant).pipe(
+              filter((relationshipRD: RemoteData<Relationship>) => hasValue(relationshipRD.payload)),
+              take(1)
+            ).subscribe((c) => {
+              this.updateAfterPatchSubmissionId = submissionId;
+              this.relationshipService.refreshRelationshipItemsInCache(item1);
+              this.relationshipService.refreshRelationshipItemsInCache(item2);
+            });
           }
         }
       )
     );
 
+  @Effect({ dispatch: false }) updateRelationshipActions$ = this.actions$
+    .pipe(
+      ofType(RelationshipActionTypes.UPDATE_RELATIONSHIP),
+      map((action: UpdateRelationshipAction) => {
+        this.updateAfterPatchSubmissionId = action.payload.submissionId;
+      })
+    );
+
+  @Effect() commitServerSyncBuffer = this.actions$
```

Comment:
Same here, I'd appreciate some TypeDocs / comments to describe this Effect.

Reply:
Added TypeDoc